### ISSUE TITLE
Remove 32bit iOS support

### ISSUE THREAD

**Issue by changkun** [fyne-io/fyne#2508]
### Is your feature request related to a problem? Please describe:

- Supporting them complicates the development process.
- It introduces many edge cases which are non-trivial to observe and easy to break.
- Rare actual users, not developers. Why do people want to deploy to built an app to a platform if the platform has very limited number of users?

### Is it possible to construct a solution with the existing API?

No

### Describe the solution you'd like to see:

Remove 32bit iOS support in the next release.

**Comment by andydotxyz** [comment]
Personally I am against this, we support Go 1.14 which supports iOS 32bit.
Apart from 2 open bugs (file dialog crash and status bar under-draw) it seems to work well.
We support Android devices which was launched 8 years ago, Apple's iOS 10.3 was the last to support 32bit and it was released around 4 years ago.

**Comment by changkun** [comment]
> Apart from 2 open bugs (file dialog crash and status bar under-draw) it seems to work well.
> We support Android KitKat which was launched 8 years ago, Apple's iOS 10.3 was the last to support 32bit and it was released around 4 years ago.

These two arguments seem relatively weak.

1. The 2 opening bugs shows people are not targeting these platform as they are important bugs
2. Supporting "X years go devices" seems to be a watchword for advertisement, but it is also a technical debt to pay, especially in long term perspective. Say for how long iOS 32bit should be supported? 6 months? 2 years? Any longer means that one have to test these combinations to guarantee it indeed works: 5 Go versions (1.14,1.15,1.16,1.17,1.18), 9 iOS platforms (ios/arm, ios/arm64, ios/amd64, darwin/arm, darwin/arm64, darwin/amd64, iossimulator/arm, iossimulator/arm64, iossimulator/amd64), that is already: 45 combinations of possibilities.

**Comment by Jacalz** [comment]
I am actually in favour for dropping the 32-bit support. Looking through all my installed applications, only three out of 30 had support for iOS 10. The vast majority supported  with iOS 13 or later, after that there were a bunch with iOS12+ and then maybe five-ish supporting iOS11+. I think it would be a reasonable target to drop, especially with iPhone 5 and lower not even making up 1% of the total marketshare https://deviceatlas.com/blog/most-popular-iphones.

**Comment by andydotxyz** [comment]
This is probably because Apple standard tools don't compile for it... So folk have to do a little extra work to support these completely functional devices. That's the thing that I am trying to push against - I don't feel that Apple pushing people off a platform means we should abandon it.
If there was a compelling reason to stop supporting it I would understand, but so far there is not a significant win that we obtain for leaving it behind.

**Comment by changkun** [comment]
I got my iPhone 5 repaired today, and after I re-explored the system for a while, here are some interesting data to show.
In Germany, the App Store top 10 free non-game apps: all require iOS 12.0. All games require iOS 12.0

- Communication apps, such as WhatsApp and Telegram, offer iOS 10 support. However, other basic infra-softawres:
- Financial-related apps, such as S-pushTAN even require iOS 13, and PayPal requires iOS 11.
- Traveling-related apps, such as Google Map, DB Navigator, Booking, etc. They all require iOS 12 or even iOS 13.
- More importantly, under the current COVID situation in Europe: All accepted software that is required by German law, such as CovPass, Corona-Warn, and Luca all require iOS 12.0.

This means the remaining users can't even use it in their daily life since they can't use it for traveling, paying, gaming but only basic IM.

Based on these data, I'd say iOS 10 is dead basically. For iPad 2, the above data also applies. Not sure why people still using the device, but it is clear that they are not using such a device daily and actively.

**Comment by andydotxyz** [comment]
I don't know that you can use the supported levels of apps inside Apple store as a measure of the OS that they want to support - as Apple dictates that.
More interesting is ad-hoc or corporate distribution (MDM) where they can support whichever device sets they desire.

**Comment by changkun** [comment]
> When we upgrade the minimum to 1.15 this conversation becomes moot.

If we are removing 1.15 in the next release, then this proposal should be accepted then? and PRs that are targeting for the next release can be accepted :)

**Comment by andydotxyz** [comment]
I don’t think we should have different minimum Go versions on different platforms.

**Comment by changkun** [comment]
I am running out of arguments. It looks like we failed to converge the discussion here, and not much can proceed on the topic. Instead of spending more energy here, I'd suggest that we move on and leave the 32bit iOS related issues open and see if there are people interested in supporting them. What do you think?
